packages/main/src/plugin/proxy.ts

```diff
@@ -22,6 +22,30 @@ import { Emitter } from './events/emitter.js';
 import { getProxyUrl } from './proxy-resolver.js';
 import { ProxyAgent } from 'undici';
 
+function ensureURL(urlstring: string | undefined): string | undefined {
+  if (urlstring) {
+    try {
+      const url = new URL(urlstring);
+      if (url.hostname) {
```

Comment:
What `urlstring' should be passed to URL constructor, to get falsy `url.hostname`?
It always has `hostname` if URL is created out of string or it throws exception.

Reply:
Yes but hostname is then empty so section is executed

Reply:
Added unit tests to check